src/workerd/api/sql.c++

```diff
@@ -18,6 +18,24 @@ jsg::Ref<SqlStorage::Cursor> SqlStorage::exec(jsg::Lock& js, kj::String querySql
   return jsg::alloc<Cursor>(*sqlite, regulator, querySql, kj::mv(bindings));
 }
 
+SqlStorage::SavepointResult SqlStorage::savepoint(jsg::Lock& js, jsg::Function<jsg::Value()> callback) {
+  // SAVEPOINT is a readonly statement, but we need to trigger an outer TRANSACTION
+  KJ_IF_MAYBE(cb, sqlite->onWriteCallback) {
+    (*cb)();
+  }
+  // TODO(sqlite) allow nested savepoints, using a new savepoint name for each depth
+  sqlite->prepare("SAVEPOINT aaa;").run();
```

Comment:
If we're not going to support nesting yet then we should have some kind of graceful failure if someone tries until this is enabled.

Reply:
I agree, although I'm actually not sure how? I'm not clear how to make track whether we're already inside a `transactionSync` call—if I could do that, we may as well track the depth to create a unique savepoint name and therefore allow nesting.